Spigot-Server-Patches/0437-Add-Heightmap-API.patch

```diff
@@ -0,0 +1,45 @@
+From 08207756e4a12b1935bed0692773e66152c753ef Mon Sep 17 00:00:00 2001
+From: Spottedleaf <Spottedleaf@users.noreply.github.com>
+Date: Tue, 1 Jan 2019 02:22:01 -0800
+Subject: [PATCH] Add Heightmap API
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+index 40ee34675..5ef95e936 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+@@ -136,6 +136,31 @@ public class CraftWorld implements World {
+         return world.getHighestBlockYAt(HeightMap.Type.LIGHT_BLOCKING, new BlockPosition(x, 0, z)).getY();
+     }
+ 
++    // Paper start - Implement heightmap api
++    @Override
++    public int getHighestBlockYAt(final int x, final int z, final HeightmapType heightmap) throws UnsupportedOperationException {
++        loadChunk(x >> 4, z >> 4);
```

Comment:
we don't need loadChunk here
replace this method to use `world.a(HeightMap.Type heightmap_type, int i, int j);`
instead and add OBF helper

Reply:
we need loadchunk since the heightmap will return 0 for unloaded chunks